stepanblyschak/sonic-snmpagent
# [rfc1213] Interface MIB add l3 vlan interfaces & aggregate rif counters

**- What I did**

Rework based on https://github.com/Azure/sonic-snmpagent/pull/133 and feedback https://github.com/Azure/sonic-snmpagent/issues/148.


**- How I did it**

- merge #133 with upstream master.
- Fix UnavailableDataError when RIF counters are not enabled.
- Make RIF keys in maps unique in multi asic env.

**- How to verify it**

- Disable rif counters
- restart swss
- snmpwalk
- Verify no UnavailableDataError in logs
- Verify port counters are returned
- Enable rif counters
- snmpwalk
- Verify no error in logs
- Verify error in/out counters are aggregated for rif ports and portchannels
- Verify VLAN RIF counters are present in the MIB.

**- Description for the changelog**

Added support for aggregated router interface counters and L3 VLAN counters to RFC1213 MIB.
